src/rpc/blockchain.cpp

```diff
@@ -1614,6 +1614,11 @@ UniValue savemempool(const JSONRPCRequest& request)
         );
     }
 
+    TxMempoolDumpReserver reserver(mempool);
+    if (!reserver.reserve()) {
+        throw JSONRPCError(RPC_MISC_ERROR, "Currently dumping the mempool");
```

Comment:
Nit: maybe instead, 

"`savemempool` already in progress"

Reply:
Sounds better.